packages/@aws-cdk/aws-appsync/test/integ.auth-apikey.ts

```diff
@@ -0,0 +1,68 @@
+import { join } from 'path';
+import { AttributeType, BillingMode, Table } from '@aws-cdk/aws-dynamodb';
+import { App, RemovalPolicy, Stack, Expires } from '@aws-cdk/core';
+import {
+  AuthorizationType,
+  GraphQLApi,
+  MappingTemplate,
+  PrimaryKey,
+  Values,
+} from '../lib';
+/*
+ * Creates an Appsync GraphQL API with API_KEY authorization.
+ * Testing for API_KEY Authorization.
+ *
+ * Stack verification steps:
+ * Deploy stack, get api-key and endpoint.
+ * Check if authorization occurs with empty get.
+ *
+ * -- bash verify.integ.auth-apikey.sh --start                      -- deploy stack               --
+ * -- aws appsync list-graphql-apis                                 -- obtain api id && endpoint  --
+ * -- aws appsync list-api-keys --api-id [API ID]                   -- obtain api key             --
+ * -- bash verify.integ.auth-apikey.sh --check [APIKEY] [ENDPOINT]  -- check if fails/success     --
+ * -- bash verify.integ.auth-apikey.sh --clean                      -- clean dependencies/stack   --
+ */
+
+const app = new App();
+const stack = new Stack(app, 'aws-appsync-integ');
+
+const api = new GraphQLApi(stack, 'Api', {
+  name: 'Integ_Test_APIKey',
+  schemaDefinitionFile: join(__dirname, 'appsync.auth.graphql'),
+  authorizationConfig: {
+    defaultAuthorization: {
+      authorizationType: AuthorizationType.API_KEY,
+      apiKeyConfig: {
+        // Generate a timestamp that's 365 days ahead, use atTimestamp so integ test doesn't fail
+        expires: Expires.atTimestamp(1626609600000),
+      },
+    },
+  },
+});
+
+const testTable = new Table(stack, 'TestTable', {
+  billingMode: BillingMode.PAY_PER_REQUEST,
+  partitionKey: {
+    name: 'id',
+    type: AttributeType.STRING,
+  },
+  removalPolicy: RemovalPolicy.DESTROY,
+});
+
+const testDS = api.addDynamoDbDataSource('testDataSource', 'Table for Tests"', testTable);
+
+testDS.createResolver({
+  typeName: 'Query',
+  fieldName: 'getTests',
+  requestMappingTemplate: MappingTemplate.dynamoDbScanTable(),
+  responseMappingTemplate: MappingTemplate.dynamoDbResultList(),
```

Comment:
How is this related to this change?

Reply:
same as the comment above

> just for the integrity test to see if the API Key works for query/mutation

Reply:
@eladb o i see.. i just need to check if the apikey exist is that right?

Reply:
i think its important to test the connection else i can just remove the integ test all together